app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java

```diff
@@ -0,0 +1,449 @@
+     * @param correct correct list of ingredients loaded from file
+     * @param extracted list of ingredients extracted by the OCR
+     * @return confidence percentage based on number of matched words, their similarity and order
+     * @author Francesco Pham credit to Stefano Romanello for Levenshtein library suggestion
+     */
+    private float ingredientsTextComparison(String correct, String extracted){
+
+        extracted = extracted.toLowerCase(); //ignoring case
+
+        //split words
+        String[] extractedWords = extracted.trim().split("[ ,-:./\\n\\r]+");
+        String[] correctWords = correct.trim().split("[ ,-:./\\n\\r]+");
+
+        Log.i(TAG, "ingredientsTextComparison -> Start of comparing");
+        Log.i(TAG, "ingredientsTextComparison -> correctWords.length == " + correctWords.length + ", extractedWords.length == " + extractedWords.length);
+
+        float points = 0;
+        int maxPoints = 0;
+        int posLastWordFound = 0;
+        int consecutiveNotFound = 0;
+
+
+        WeightedLevenshtein levenshtein = new WeightedLevenshtein(
+                new CharacterSubstitutionInterface() {
+                    public double cost(char c1, char c2) {
+
+                        // The cost for substituting 't' and 'r' is considered
+                        // smaller as these 2 are located next to each other
+                        // on a keyboard
+                        if (c1 == 't' && c2 == 'r') {
+                            return 0.5;
+                        }
+                        else if (c1 == 'q' && c2 == 'o') {
+                            return 0.5;
+                        }
+                        else if (c1 == 'I' && c2 == 'l') {
+                            return 0.5;
+                        }
+
+                        // For most cases, the cost of substituting 2 characters
+                        // is 1.0
+                        return 1.0;
+                    }
+                });
+
+        //for each correct word
+        for (String word : correctWords) {
+            boolean found = false;
+            int index = posLastWordFound;
+            word = word.toLowerCase(); //ignoring case
+
+            if (word.length() >= 3) { //ignoring non significant words with 1 or 2 characters
+                maxPoints += word.length();
+
+                for (int i = 0; i < extractedWords.length && !found; i++) {
+                    //for each extracted words starting from posLastWordFound
+                    index = (posLastWordFound + i) % extractedWords.length;
+
+                    //Calculate similarity and normalize
+                    int maxLength = Math.max(word.length(),extractedWords[index].length());
+                    double similarity = 1.0 - levenshtein.distance(word,extractedWords[index])/maxLength;
+
+                    //if similarity grater than 0.8 the word is found
+                    if (similarity > 0.8) {
```

Comment:
I do not understand why> 0.8. Is it a parameter that you gave yourself or is it default?

Reply:
I made it a final variable in commit 56d3cb4